internal/pkg/cli/local_run.go

```diff
@@ -323,12 +346,17 @@ func getBuiltSidecarImageLocations(sidecars map[string]*manifest.SidecarConfig)
 	return sideCarBuiltImageLocations
 }
 
-func (o *localRunOpts) runPauseContainer(ctx context.Context, containerPorts map[string]string) error {
+func (o *localRunOpts) runPauseContainer(ctx context.Context, ports map[string]string) error {
+	// flip ports to be host->ctr
+	flippedPorts := make(map[string]string, len(ports))
+	for k, v := range ports {
+		flippedPorts[v] = k
+	}
```

Comment:
Why do we flip ports?

Reply:
docker expects ports in the format `host port:container port`, but we _store_ them in the format of `container port -> host port` so that applying port overrides is easier.

say my task def exposes `80:7000` (80 host port, 7000 container port). If I want to override that to expose `:8000` instead of `:80` on my laptop, I would use `--port-override 8000:7000`. Because we store `ports` with a key of the container port, this is easy, just `ports[7000] = 8000`. Otherwise, we'd have to go look through a map of `host -> container` and delete any entries where the container port is 7000 before inserting the mapping.

though now i'm wondering if we need to consider the case where there are intentionally multiple host ports for a single container port 🤔 i.e., `8000:80`, `7000:80`, etc.

Reply:
It is a possible scenario i think. And they'll be surprised if we don't allow them to do that.

Reply:
actually! i just tested a few things with ECS to verify, and I don't think this is a use case supported by ECS, at least while using "awsvpc," due to this restriction https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-taskdefinition-portmapping.html#cfn-ecs-taskdefinition-portmapping-hostport.

Reply:
Speaking of `awsvpc` mode, is it possible for them to have "`80:7000` (80 host port, 7000 container port)"? I thought host port and container port were always the same https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_PortMapping.html

Reply:
yeah, you're right - it's not possible for that to exist 😊 but, we do want to allow them to override that port. so a good example would be:
taskdef: 80 host, 80 container
override: `--override-port 8080:80` (8080 on laptop, 80 in container)

Reply:
Yeah if that is the case i would expect `8000:80`, `7000:80` to work as well i guess 🤔 Or we can easily get this around by only allowing one override for now?

Reply:
yeah i think one override per container port is fine for now, until we see a use case where exposing the same port on different host ports makes sense - especially considering that's _not_ a case that is supported with `awsvpc`

Reply:
Yeah it's backwardcompatible anyway - ux will be the same